src/string_search.h

```diff
@@ -274,111 +229,133 @@ inline uint8_t GetHighestValueByte(uint16_t character) {
 inline uint8_t GetHighestValueByte(uint8_t character) { return character; }
 
 
-template <typename PatternChar, typename SubjectChar>
-inline size_t FindFirstCharacter(Vector<const PatternChar> pattern,
-                              Vector<const SubjectChar> subject, size_t index) {
-  const PatternChar pattern_first_char = pattern[0];
+// Searches for a byte value in a memory buffer, back to front
+// Uses memrchr(3) on systems which support it, for speed
+// Falls back to a vanilla for loop on systems which don't
+inline const void* MemrchrFill(const void* haystack, uint8_t needle,
+                               size_t haystack_len) {
+#ifdef _GNU_SOURCE
```

Comment:
`_GNU_SOURCE` is included by default when compiling against glibc, which I think we are on Mac and Linux.

http://stackoverflow.com/questions/11670581/why-is-gnu-source-defined-by-default-and-how-to-turn-it-off

I just tested it & confirmed it's defined when I run `make`. I'm using Ubuntu 15.10.

I think we want to keep using the built-in `memrchr` when available because it's faster than a simple for loop: http://cebka.blogspot.com/2015/04/how-fast-is-your-memchr.html